java-squid/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValue.java

```diff
@@ -152,156 +157,7 @@ boolean isPredefined() {
-          }
-          Boolean fulfilled = programState.isFulfilled(relationToFulfill);
-          if (fulfilled == null) {
-            programState = programState.addRelation(relationToFulfill);
-          } else if (!fulfilled.booleanValue()) {
-            return null;
-          }
-        }
-      }
-      return programState;
-    }
-
-    @CheckForNull
-    protected SymbolicValueRelation getSymbolicValueRelation() {
-      return null;
-    }
-
-    @Override
-    public String toString() {
-      return "EQ_TO_" + super.toString();
-    }
-
-    private List<ProgramState> copyConstraint(SymbolicValue from, SymbolicValue to, ProgramState programState, BooleanConstraint booleanConstraint) {
-      Object constraintLeft = programState.getConstraint(from);
-      if (constraintLeft instanceof BooleanConstraint) {
-        BooleanConstraint boolConstraint = (BooleanConstraint) constraintLeft;
-        return to.setConstraint(programState, shouldNotInverse().equals(booleanConstraint) ? boolConstraint : boolConstraint.inverse());
-      } else if (constraintLeft instanceof ObjectConstraint) {
-        ObjectConstraint nullConstraint = (ObjectConstraint) constraintLeft;
-        if (nullConstraint.equals(ObjectConstraint.NULL)) {
-          return to.setConstraint(programState, shouldNotInverse().equals(booleanConstraint) ? nullConstraint : nullConstraint.inverse());
-        } else if (shouldNotInverse().equals(booleanConstraint)) {
-          return to.setConstraint(programState, nullConstraint);
-        }
-      }
-      return ImmutableList.of(programState);
-    }
-
-  }
-
-  static class NotEqualToSymbolicValue extends BinarySymbolicValue {
-
-    public NotEqualToSymbolicValue(int id) {
-      super(id);
-    }
-
-    @Override
-    public String toString() {
-      return "NEQ_TO_" + super.toString();
-    }
-
-    @Override
-    BooleanConstraint shouldNotInverse() {
-      return BooleanConstraint.FALSE;
-    }
-
-    @Override
-    @CheckForNull
-    protected SymbolicValueRelation getSymbolicValueRelation() {
-      return new NotEqualRelation(leftOp, rightOp);
-    }
-  }
-
-  static class EqualToSymbolicValue extends BinarySymbolicValue {
-
-    public EqualToSymbolicValue(int id) {
-      super(id);
-    }
-
-    @Override
-    BooleanConstraint shouldNotInverse() {
-      return BooleanConstraint.TRUE;
-    }
-
-    @Override
-    @CheckForNull
-    protected SymbolicValueRelation getSymbolicValueRelation() {
-      return new EqualRelation(leftOp, rightOp);
-    }
-  }
-
-  static class MethodEqualsToSymbolicValue extends EqualToSymbolicValue {
-
-    public MethodEqualsToSymbolicValue(int id) {
-      super(id);
-    }
-
-    @Override
-    protected SymbolicValueRelation getSymbolicValueRelation() {
-      return new MethodEqualsRelation(leftOp, rightOp);
-    }
-  }
-
-  abstract static class UnarySymbolicValue extends SymbolicValue {
+  public abstract static class UnarySymbolicValue extends SymbolicValue {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Override this superclass' "equals" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2160)